cmd/updater/update.sh

```diff
@@ -330,14 +330,27 @@ function validate_update() {
     return 0
 }
 
+function run_systemd_action() {
+    local action=$1
+    local data_dir=$2
+
+    # Get the owner of a running process.
+    owner=$(awk '{ print $1 }' <(ps aux | grep "[a]lgod"))
```

Comment:
the grep here is a bit weak predicate.. if could match other processes that you didn't mean.. maybe do some extra filtering in the awk ?

Reply:
I can see if it matches the data directory as well.